enzyme/Enzyme/MLIR/Interfaces/GradientUtilsReverse.h

```diff
@@ -0,0 +1,120 @@
+//===- GradientUtilsReverse.h - Utilities for gradient interfaces -------* C++
+//-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/IR/BlockAndValueMapping.h"
+#include "mlir/IR/FunctionInterfaces.h"
+
```

Comment:
Can GradientUtils and GradientUtilsReverse be fused?

Reply:
I currently don't feel it is really feasible to merge GradientUtils and GradientUtilsReverse, since both files currently share very little code. Currently, I would suggest slowly extracting a common core (such getNewFromOriginal) between both classes, to maintain readability of the code for now.